File: src/Azdo.Tui/Styles/Themes.cs
using System.Text.Json;
using System.Text.Json.Serialization;

namespace Azdo.Tui.Styles;

/// <summary>
/// Built-in theme registry plus custom-theme loading (≈ <c>styles/themes.go</c>).
/// </summary>
public static class Themes
{
    private static readonly Dictionary<string, Theme> Registry;

    static Themes()
    {
        // Built in a static constructor so the built-in theme fields (declared
        // below) are guaranteed initialized before they're inserted.
        Registry = new Dictionary<string, Theme>
        {
            ["dark"] = Dark,
            ["gruvbox"] = Gruvbox,
            ["nord"] = Nord,
            ["dracula"] = Dracula,
            ["catppuccin"] = Catppuccin,
            ["github"] = GitHub,
            ["retro"] = Retro,
            ["monokai"] = Monokai,
        };
    }

    /// <summary>Returns a theme by name, or throws <see cref="ThemeException"/> if absent.</summary>
    public static Theme GetByName(string? name)
    {
        if (string.IsNullOrEmpty(name) || !Registry.TryGetValue(name, out var t))
            throw new ThemeException("theme not found");
        return t;
    }

    public static bool TryGetByName(string? name, out Theme theme)
    {
        if (!string.IsNullOrEmpty(name) && Registry.TryGetValue(name, out var t))
        {
            theme = t;
            return true;
        }
        theme = Default;
        return false;
    }

    public static Theme GetByNameWithFallback(string name) => TryGetByName(name, out var t) ? t : Default;

    /// <summary>The default theme is Dracula.</summary>
    public static Theme Default => Dracula;

    public static IReadOnlyList<string> ListAvailable() => Registry.Keys.OrderBy(k => k, StringComparer.Ordinal).ToList();

    public static string GetThemesDirectoryPath()
    {
        var home = Environment.GetFolderPath(Environment.SpecialFolder.UserProfile);
        return Path.Combine(home, ".config", "azdo-tui", "themes");
    }

    /// <summary>Registers a custom theme after validation.</summary>
    public static void Register(Theme theme)
    {
        theme.Validate();
        Registry[theme.Name] = theme;
    }

    private sealed class ThemeJson
    {
        [JsonPropertyName("name")] public string Name { get; set; } = "";
        [JsonPropertyName("primary")] public string Primary { get; set; } = "";
        [JsonPropertyName("secondary")] public string Secondary { get; set; } = "";
        [JsonPropertyName("accent")] public string Accent { get; set; } = "";
        [JsonPropertyName("success")] public string Success { get; set; } = "";
        [JsonPropertyName("warning")] public string Warning { get; set; } = "";
        [JsonPropertyName("error")] public string Error { get; set; } = "";
        [JsonPropertyName("info")] public string Info { get; set; } = "";
        [JsonPropertyName("background")] public string Background { get; set; } = "";
        [JsonPropertyName("background_alt")] public string BackgroundAlt { get; set; } = "";
        [JsonPropertyName("background_select")] public string BackgroundSelect { get; set; } = "";
        [JsonPropertyName("foreground")] public string Foreground { get; set; } = "";
        [JsonPropertyName("foreground_muted")] public string ForegroundMuted { get; set; } = "";
        [JsonPropertyName("foreground_bold")] public string ForegroundBold { get; set; } = "";
        [JsonPropertyName("select_foreground")] public string SelectForeground { get; set; } = "";
        [JsonPropertyName("select_background")] public string SelectBackground { get; set; } = "";
        [JsonPropertyName("border")] public string Border { get; set; } = "";
        [JsonPropertyName("link")] public string Link { get; set; } = "";
        [JsonPropertyName("spinner")] public string Spinner { get; set; } = "";
        [JsonPropertyName("tab_active_foreground")] public string TabActiveForeground { get; set; } = "";
        [JsonPropertyName("tab_active_background")] public string TabActiveBackground { get; set; } = "";
        [JsonPropertyName("tab_inactive_foreground")] public string TabInactiveForeground { get; set; } = "";
        [JsonPropertyName("tab_inactive_background")] public string TabInactiveBackground { get; set; } = "";
    }

    /// <summary>Parses a theme from JSON; throws on invalid JSON or validation failure.</summary>
    public static Theme LoadFromJson(string json)
    {
        var tj = JsonSerializer.Deserialize<ThemeJson>(json) ?? throw new ThemeException("failed to parse theme JSON");
        var theme = new Theme
        {
            Name = tj.Name, Primary = tj.Primary, Secondary = tj.Secondary, Accent = tj.Accent,
            Success = tj.Success, Warning = tj.Warning, Error = tj.Error, Info = tj.Info,
            Background = tj.Background, BackgroundAlt = tj.BackgroundAlt, BackgroundSelect = tj.BackgroundSelect,
            Foreground = tj.Foreground, ForegroundMuted = tj.ForegroundMuted, ForegroundBold = tj.ForegroundBold,
            SelectForeground = tj.SelectForeground, SelectBackground = tj.SelectBackground,
            Border = tj.Border, Link = tj.Link, Spinner = tj.Spinner,
            TabActiveForeground = tj.TabActiveForeground, TabActiveBackground = tj.TabActiveBackground,
            TabInactiveForeground = tj.TabInactiveForeground, TabInactiveBackground = tj.TabInactiveBackground,
        };
        theme.Validate();
        return theme;
    }

    /// <summary>Loads all <c>*.json</c> themes from a directory; invalid files are skipped.</summary>
    public static int LoadCustomThemesFromDirectory(string dir)
    {
        if (!Directory.Exists(dir)) return 0;
        int loaded = 0;
        foreach (var file in Directory.EnumerateFiles(dir, "*.json"))
        {
            try
            {
                var theme = LoadFromJson(File.ReadAllText(file));
                Register(theme);
                loaded++;
            }
            catch { /* skip invalid theme files */ }
        }
        return loaded;
    }

    public static readonly Theme Dark = new()
    {
        Name = "dark",
        Primary = "33", Secondary = "39", Accent = "212",
        Success = "42", Warning = "214", Error = "196", Info = "33",
        Background = "236", BackgroundAlt = "235", BackgroundSelect = "57",
        Foreground = "252", ForegroundMuted = "243", ForegroundBold = "255",
        SelectForeground = "229", SelectBackground = "57",
        Border = "240", Link = "81", Spinner = "205",
        TabActiveForeground = "229", TabActiveBackground = "57", TabInactiveForeground = "252",
    };

    public static readonly Theme Gruvbox = new()
    {
        Name = "gruvbox",
        Primary = "#458588", Secondary = "#689d6a", Accent = "#d3869b",
        Success = "#b8bb26", Warning = "#fabd2f", Error = "#fb4934", Info = "#83a598",
        Background = "#282828", BackgroundAlt = "#1d2021", BackgroundSelect = "#3c3836",
        Foreground = "#ebdbb2", ForegroundMuted = "#928374", ForegroundBold = "#fbf1c7",
        SelectForeground = "#fabd2f", SelectBackground = "#504945",
        Border = "#504945", Link = "#83a598", Spinner = "#d3869b",
        TabActiveForeground = "#fabd2f", TabActiveBackground = "#504945", TabInactiveForeground = "#a89984",
    };

    public static readonly Theme Nord = new()
    {
        Name = "nord",
        Primary = "#81a1c1", Secondary = "#88c0d0", Accent = "#b48ead",
        Success = "#a3be8c", Warning = "#ebcb8b", Error = "#bf616a", Info = "#5e81ac",
        Background = "#2e3440", BackgroundAlt = "#3b4252", BackgroundSelect = "#434c5e",
        Foreground = "#eceff4", ForegroundMuted = "#4c566a", ForegroundBold = "#eceff4",
        SelectForeground = "#eceff4", SelectBackground = "#434c5e",
        Border = "#4c566a", Link = "#88c0d0", Spinner = "#b48ead",
        TabActiveForeground = "#eceff4", TabActiveBackground = "#5e81ac", TabInactiveForeground = "#d8dee9",
    };

    public static readonly Theme Dracula = new()
    {
        Name = "dracula",
        Primary = "#bd93f9", Secondary = "#8be9fd", Accent = "#ff79c6",
        Success = "#50fa7b", Warning = "#f1fa8c", Error = "#ff5555", Info = "#8be9fd",
        Background = "#282a36", BackgroundAlt = "#21222c", BackgroundSelect = "#44475a",
        Foreground = "#f8f8f2", ForegroundMuted = "#6272a4", ForegroundBold = "#f8f8f2",
        SelectForeground = "#f8f8f2", SelectBackground = "#44475a",
        Border = "#6272a4", Link = "#8be9fd", Spinner = "#ff79c6",
        TabActiveForeground = "#f8f8f2", TabActiveBackground = "#bd93f9", TabInactiveForeground = "#f8f8f2",
    };

    public static readonly Theme Catppuccin = new()
    {
        Name = "catppuccin",
        Primary = "#89b4fa", Secondary = "#94e2d5", Accent = "#cba6f7",
        Success = "#a6e3a1", Warning = "#f9e2af", Error = "#f38ba8", Info = "#89dceb",
        Background = "#1e1e2e", BackgroundAlt = "#181825", BackgroundSelect = "#313244",
        Foreground = "#cdd6f4", ForegroundMuted = "#6c7086", ForegroundBold = "#cdd6f4",
        SelectForeground = "#cdd6f4", SelectBackground = "#45475a",
        Border = "#585b70", Link = "#89dceb", Spinner = "#f5c2e7",
        TabActiveForeground = "#1e1e2e", TabActiveBackground = "#89b4fa", TabInactiveForeground = "#bac2de",
    };

    public static readonly Theme GitHub = new()
    {
        Name = "github",
        Primary = "#58a6ff", Secondary = "#56d4dd", Accent = "#bc8cff",
        Success = "#3fb950", Warning = "#d29922", Error = "#f85149", Info = "#58a6ff",
        Background = "#0d1117", BackgroundAlt = "#161b22", BackgroundSelect = "#21262d",
        Foreground = "#c9d1d9", ForegroundMuted = "#8b949e", ForegroundBold = "#f0f6fc",
        SelectForeground = "#f0f6fc", SelectBackground = "#264f78",
        Border = "#30363d", Link = "#58a6ff", Spinner = "#bc8cff",
        TabActiveForeground = "#f0f6fc", TabActiveBackground = "#58a6ff", TabInactiveForeground = "#8b949e",
    };

    public static readonly Theme Retro = new()
    {
        Name = "retro",
        Primary = "#00ff41", Secondary = "#00cc33", Accent = "#39ff14",
        Success = "#00ff41", Warning = "#ccff00", Error = "#ff003c", Info = "#00cc33",
        Background = "#0a0a0a", BackgroundAlt = "#050505", BackgroundSelect = "#003300",
        Foreground = "#00ff41", ForegroundMuted = "#336633", ForegroundBold = "#66ff66",
        SelectForeground = "#0a0a0a", SelectBackground = "#00ff41",
        Border = "#004400", Link = "#39ff14", Spinner = "#00ff41",
        TabActiveForeground = "#0a0a0a", TabActiveBackground = "#00ff41", TabInactiveForeground = "#336633",
    };

    public static readonly Theme Monokai = new()
    {
        Name = "monokai",
        Primary = "#66d9ef", Secondary = "#a6e22e", Accent = "#f92672",
        Success = "#a6e22e", Warning = "#e6db74", Error = "#f92672", Info = "#66d9ef",
        Background = "#272822", BackgroundAlt = "#1e1f1c", BackgroundSelect = "#49483e",
        Foreground = "#f8f8f2", ForegroundMuted = "#75715e", ForegroundBold = "#f8f8f2",
        SelectForeground = "#f8f8f2", SelectBackground = "#49483e",
        Border = "#75715e", Link = "#66d9ef", Spinner = "#ae81ff",
        TabActiveForeground = "#f8f8f2", TabActiveBackground = "#f92672", TabInactiveForeground = "#75715e",
    };
}

File: reference/internal/ui/styles/themes.go
package styles

import (
	"encoding/json"
	"fmt"
	"os"
	"path/filepath"
	"sort"

	"github.com/charmbracelet/lipgloss"
)

// themeRegistry holds all built-in themes
var themeRegistry = map[string]Theme{
	"dark":       darkTheme,
	"gruvbox":    gruvboxTheme,
	"nord":       nordTheme,
	"dracula":    draculaTheme,
	"catppuccin": catppuccinTheme,
	"github":     githubTheme,
	"retro":      retroTheme,
	"monokai":    monokaiTheme,
}

// GetThemeByName returns a theme by name.
// Returns ErrThemeNotFound if the theme doesn't exist.
func GetThemeByName(name string) (Theme, error) {
	if name == "" {
		return Theme{}, ErrThemeNotFound
	}
	theme, ok := themeRegistry[name]
	if !ok {
		return Theme{}, ErrThemeNotFound
	}
	return theme, nil
}

// GetThemeByNameWithFallback returns a theme by name, falling back to the default
// theme if the requested theme doesn't exist.
func GetThemeByNameWithFallback(name string) Theme {
	theme, err := GetThemeByName(name)
	if err != nil {
		return GetDefaultTheme()
	}
	return theme
}

// GetDefaultTheme returns the default dracula theme.
func GetDefaultTheme() Theme {
	return draculaTheme
}

// ListAvailableThemes returns a sorted list of all available theme names.
func ListAvailableThemes() []string {
	names := make([]string, 0, len(themeRegistry))
	for name := range themeRegistry {
		names = append(names, name)
	}
	sort.Strings(names)
	return names
}

// GetThemesDirectoryPath returns the path to the custom themes directory.
// Custom themes should be placed in ~/.config/azdo-tui/themes/
func GetThemesDirectoryPath() (string, error) {
	homeDir, err := os.UserHomeDir()
	if err != nil {
		return "", fmt.Errorf("failed to get home directory: %w", err)
	}
	return filepath.Join(homeDir, ".config", "azdo-tui", "themes"), nil
}

// themeJSON represents the JSON structure for custom themes.
// Uses snake_case for JSON field names to match common conventions.
type themeJSON struct {
	Name                  string `json:"name"`
	Primary               string `json:"primary"`
	Secondary             string `json:"secondary"`
	Accent                string `json:"accent"`
	Success               string `json:"success"`
	Warning               string `json:"warning"`
	Error                 string `json:"error"`
	Info                  string `json:"info"`
	Background            string `json:"background"`
	BackgroundAlt         string `json:"background_alt"`
	BackgroundSelect      string `json:"background_select"`
	Foreground            string `json:"foreground"`
	ForegroundMuted       string `json:"foreground_muted"`
	ForegroundBold        string `json:"foreground_bold"`
	SelectForeground      string `json:"select_foreground"`
	SelectBackground      string `json:"select_background"`
	Border                string `json:"border"`
	Link                  string `json:"link"`
	Spinner               string `json:"spinner"`
	TabActiveForeground   string `json:"tab_active_foreground"`
	TabActiveBackground   string `json:"tab_active_background"`
	TabInactiveForeground string `json:"tab_inactive_foreground"`
	TabInactiveBackground string `json:"tab_inactive_background"`
}

// LoadThemeFromJSON loads a theme from JSON bytes.
// Returns an error if the JSON is invalid or the theme fails validation.
func LoadThemeFromJSON(data []byte) (Theme, error) {
	var tj themeJSON
	if err := json.Unmarshal(data, &tj); err != nil {
		return Theme{}, fmt.Errorf("failed to parse theme JSON: %w", err)
	}

	// Convert JSON structure to Theme
	theme := Theme{
		Name:                  tj.Name,
		Primary:               lipgloss.Color(tj.Primary),
		Secondary:             lipgloss.Color(tj.Secondary),
		Accent:                lipgloss.Color(tj.Accent),
		Success:               lipgloss.Color(tj.Success),
		Warning:               lipgloss.Color(tj.Warning),
		Error:                 lipgloss.Color(tj.Error),
		Info:                  lipgloss.Color(tj.Info),
		Background:            lipgloss.Color(tj.Background),
		BackgroundAlt:         lipgloss.Color(tj.BackgroundAlt),
		BackgroundSelect:      lipgloss.Color(tj.BackgroundSelect),
		Foreground:            lipgloss.Color(tj.Foreground),
		ForegroundMuted:       lipgloss.Color(tj.ForegroundMuted),
		ForegroundBold:        lipgloss.Color(tj.ForegroundBold),
		SelectForeground:      lipgloss.Color(tj.SelectForeground),
		SelectBackground:      lipgloss.Color(tj.SelectBackground),
		Border:                lipgloss.Color(tj.Border),
		Link:                  lipgloss.Color(tj.Link),
		Spinner:               lipgloss.Color(tj.Spinner),
		TabActiveForeground:   lipgloss.Color(tj.TabActiveForeground),
		TabActiveBackground:   lipgloss.Color(tj.TabActiveBackground),
		TabInactiveForeground: lipgloss.Color(tj.TabInactiveForeground),
		TabInactiveBackground: lipgloss.Color(tj.TabInactiveBackground),
	}

	// Validate the theme
	if err := theme.Validate(); err != nil {
		return Theme{}, fmt.Errorf("invalid theme: %w", err)
	}

	return theme, nil
}

// RegisterTheme registers a custom theme in the theme registry.
// Returns an error if the theme fails validation.
func RegisterTheme(theme Theme) error {
	if err := theme.Validate(); err != nil {
		return fmt.Errorf("invalid theme: %w", err)
	}
	themeRegistry[theme.Name] = theme
	return nil
}

// LoadCustomThemesFromDirectory loads all custom theme files from a directory.
// Theme files must have a .json extension and contain valid theme JSON.
// Returns the number of themes successfully loaded and any critical errors.
// Missing directory or individual invalid theme files are not considered errors.
func LoadCustomThemesFromDirectory(dir string) (int, error) {
	// Check if directory exists
	if _, err := os.Stat(dir); os.IsNotExist(err) {
		// Directory doesn't exist - not an error, just no custom themes
		return 0, nil
	} else if err != nil {
		return 0, fmt.Errorf("failed to access themes directory: %w", err)
	}

	// Read all files in the directory
	entries, err := os.ReadDir(dir)
	if err != nil {
		return 0, fmt.Errorf("failed to read themes directory: %w", err)
	}

	loadedCount := 0
	for _, entry := range entries {
		// Skip directories and non-JSON files
		if entry.IsDir() || filepath.Ext(entry.Name()) != ".json" {
			continue
		}

		// Read the theme file
		themePath := filepath.Join(dir, entry.Name())
		data, err := os.ReadFile(themePath)
		if err != nil {
			// Skip files that can't be read
			continue
		}

		// Try to load the theme
		theme, err := LoadThemeFromJSON(data)
		if err != nil {
			// Skip invalid theme files
			continue
		}

		// Register the theme
		if err := RegisterTheme(theme); err != nil {
			// Skip themes that fail registration
			continue
		}

		loadedCount++
	}

	return loadedCount, nil
}

// Dark theme - the default theme matching current application colors
var darkTheme = Theme{
	Name: "dark",

	// Primary colors
	Primary:   lipgloss.Color("33"),  // Blue - headers, active status
	Secondary: lipgloss.Color("39"),  // Cyan - org/project info
	Accent:    lipgloss.Color("212"), // Magenta - keybindings, section titles

	// Status colors
	Success: lipgloss.Color("42"),  // Green
	Warning: lipgloss.Color("214"), // Orange
	Error:   lipgloss.Color("196"), // Red
	Info:    lipgloss.Color("33"),  // Blue

	// Background colors
	Background:       lipgloss.Color("236"), // Dark gray
	BackgroundAlt:    lipgloss.Color("235"), // Slightly darker
	BackgroundSelect: lipgloss.Color("57"),  // Dark blue for selection

	// Foreground colors
	Foreground:      lipgloss.Color("252"), // Light gray - main text
	ForegroundMuted: lipgloss.Color("243"), // Gray - disabled/metadata
	ForegroundBold:  lipgloss.Color("255"), // White - emphasized

	// Selection colors
	SelectForeground: lipgloss.Color("229"), // Yellow
	SelectBackground: lipgloss.Color("57"),  // Dark blue

	// UI elements
	Border:  lipgloss.Color("240"), // Gray border
	Link:    lipgloss.Color("81"),  // Blue - hyperlinks
	Spinner: lipgloss.Color("205"), // Magenta

	// Tab colors
	TabActiveForeground:   lipgloss.Color("229"), // Yellow
	TabActiveBackground:   lipgloss.Color("57"),  // Dark blue
	TabInactiveForeground: lipgloss.Color("252"), // Light gray
}

// Gruvbox dark theme
var gruvboxTheme = Theme{
	Name: "gruvbox",

	// Gruvbox palette
	Primary:   lipgloss.Color("#458588"), // Blue
	Secondary: lipgloss.Color("#689d6a"), // Aqua
	Accent:    lipgloss.Color("#d3869b"), // Purple

	// Status colors
	Success: lipgloss.Color("#b8bb26"), // Green
	Warning: lipgloss.Color("#fabd2f"), // Yellow
	Error:   lipgloss.Color("#fb4934"), // Red
	Info:    lipgloss.Color("#83a598"), // Light blue

	// Background colors
	Background:       lipgloss.Color("#282828"), // bg0
	BackgroundAlt:    lipgloss.Color("#1d2021"), // bg0_h
	BackgroundSelect: lipgloss.Color("#3c3836"), // bg1

	// Foreground colors
	Foreground:      lipgloss.Color("#ebdbb2"), // fg
	ForegroundMuted: lipgloss.Color("#928374"), // gray
	ForegroundBold:  lipgloss.Color("#fbf1c7"), // fg0

	// Selection colors
	SelectForeground: lipgloss.Color("#fabd2f"), // Yellow
	SelectBackground: lipgloss.Color("#504945"), // bg2

	// UI elements
	Border:  lipgloss.Color("#504945"), // bg2
	Link:    lipgloss.Color("#83a598"), // Light blue
	Spinner: lipgloss.Color("#d3869b"), // Purple

	// Tab colors
	TabActiveForeground:   lipgloss.Color("#fabd2f"), // Yellow
	TabActiveBackground:   lipgloss.Color("#504945"), // bg2
	TabInactiveForeground: lipgloss.Color("#a89984"), // fg4
}

// Nord theme
var nordTheme = Theme{
	Name: "nord",

	// Nord palette
	Primary:   lipgloss.Color("#81a1c1"), // Nord9 - blue
	Secondary: lipgloss.Color("#88c0d0"), // Nord8 - cyan
	Accent:    lipgloss.Color("#b48ead"), // Nord15 - purple

	// Status colors
	Success: lipgloss.Color("#a3be8c"), // Nord14 - green
	Warning: lipgloss.Color("#ebcb8b"), // Nord13 - yellow
	Error:   lipgloss.Color("#bf616a"), // Nord11 - red
	Info:    lipgloss.Color("#5e81ac"), // Nord10 - dark blue

	// Background colors
	Background:       lipgloss.Color("#2e3440"), // Nord0
	BackgroundAlt:    lipgloss.Color("#3b4252"), // Nord1
	BackgroundSelect: lipgloss.Color("#434c5e"), // Nord2

	// Foreground colors
	Foreground:      lipgloss.Color("#eceff4"), // Nord6
	ForegroundMuted: lipgloss.Color("#4c566a"), // Nord3
	ForegroundBold:  lipgloss.Color("#eceff4"), // Nord6

	// Selection colors
	SelectForeground: lipgloss.Color("#eceff4"), // Nord6
	SelectBackground: lipgloss.Color("#434c5e"), // Nord2

	// UI elements
	Border:  lipgloss.Color("#4c566a"), // Nord3
	Link:    lipgloss.Color("#88c0d0"), // Nord8
	Spinner: lipgloss.Color("#b48ead"), // Nord15

	// Tab colors
	TabActiveForeground:   lipgloss.Color("#eceff4"), // Nord6
	TabActiveBackground:   lipgloss.Color("#5e81ac"), // Nord10
	TabInactiveForeground: lipgloss.Color("#d8dee9"), // Nord4
}

// Dracula theme
var draculaTheme = Theme{
	Name: "dracula",

	// Dracula palette
	Primary:   lipgloss.Color("#bd93f9"), // Purple
	Secondary: lipgloss.Color("#8be9fd"), // Cyan
	Accent:    lipgloss.Color("#ff79c6"), // Pink

	// Status colors
	Success: lipgloss.Color("#50fa7b"), // Green
	Warning: lipgloss.Color("#f1fa8c"), // Yellow
	Error:   lipgloss.Color("#ff5555"), // Red
	Info:    lipgloss.Color("#8be9fd"), // Cyan

	// Background colors
	Background:       lipgloss.Color("#282a36"), // Background
	BackgroundAlt:    lipgloss.Color("#21222c"), // Darker background
	BackgroundSelect: lipgloss.Color("#44475a"), // Current line

	// Foreground colors
	Foreground:      lipgloss.Color("#f8f8f2"), // Foreground
	ForegroundMuted: lipgloss.Color("#6272a4"), // Comment
	ForegroundBold:  lipgloss.Color("#f8f8f2"), // Foreground

	// Selection colors
	SelectForeground: lipgloss.Color("#f8f8f2"), // Foreground
	SelectBackground: lipgloss.Color("#44475a"), // Current line

	// UI elements
	Border:  lipgloss.Color("#6272a4"), // Comment
	Link:    lipgloss.Color("#8be9fd"), // Cyan
	Spinner: lipgloss.Color("#ff79c6"), // Pink

	// Tab colors
	TabActiveForeground:   lipgloss.Color("#f8f8f2"), // Foreground
	TabActiveBackground:   lipgloss.Color("#bd93f9"), // Purple
	TabInactiveForeground: lipgloss.Color("#f8f8f2"), // Foreground
}

// Catppuccin Mocha theme
var catppuccinTheme = Theme{
	Name: "catppuccin",

	// Catppuccin Mocha palette
	Primary:   lipgloss.Color("#89b4fa"), // Blue
	Secondary: lipgloss.Color("#94e2d5"), // Teal
	Accent:    lipgloss.Color("#cba6f7"), // Mauve

	// Status colors
	Success: lipgloss.Color("#a6e3a1"), // Green
	Warning: lipgloss.Color("#f9e2af"), // Yellow
	Error:   lipgloss.Color("#f38ba8"), // Red
	Info:    lipgloss.Color("#89dceb"), // Sky

	// Background colors
	Background:       lipgloss.Color("#1e1e2e"), // Base
	BackgroundAlt:    lipgloss.Color("#181825"), // Mantle
	BackgroundSelect: lipgloss.Color("#313244"), // Surface0

	// Foreground colors
	Foreground:      lipgloss.Color("#cdd6f4"), // Text
	ForegroundMuted: lipgloss.Color("#6c7086"), // Overlay0
	ForegroundBold:  lipgloss.Color("#cdd6f4"), // Text

	// Selection colors
	SelectForeground: lipgloss.Color("#cdd6f4"), // Text
	SelectBackground: lipgloss.Color("#45475a"), // Surface1

	// UI elements
	Border:  lipgloss.Color("#585b70"), // Surface2
	Link:    lipgloss.Color("#89dceb"), // Sky
	Spinner: lipgloss.Color("#f5c2e7"), // Pink

	// Tab colors
	TabActiveForeground:   lipgloss.Color("#1e1e2e"), // Base
	TabActiveBackground:   lipgloss.Color("#89b4fa"), // Blue
	TabInactiveForeground: lipgloss.Color("#bac2de"), // Subtext1
}

// GitHub Dark theme
var githubTheme = Theme{
	Name: "github",

	// GitHub Dark palette
	Primary:   lipgloss.Color("#58a6ff"), // Blue
	Secondary: lipgloss.Color("#56d4dd"), // Cyan
	Accent:    lipgloss.Color("#bc8cff"), // Purple

	// Status colors
	Success: lipgloss.Color("#3fb950"), // Green
	Warning: lipgloss.Color("#d29922"), // Yellow/Orange
	Error:   lipgloss.Color("#f85149"), // Red
	Info:    lipgloss.Color("#58a6ff"), // Blue

	// Background colors
	Background:       lipgloss.Color("#0d1117"), // Default background
	BackgroundAlt:    lipgloss.Color("#161b22"), // Darker background
	BackgroundSelect: lipgloss.Color("#21262d"), // Selection

	// Foreground colors
	Foreground:      lipgloss.Color("#c9d1d9"), // Default text
	ForegroundMuted: lipgloss.Color("#8b949e"), // Muted text
	ForegroundBold:  lipgloss.Color("#f0f6fc"), // Bright text

	// Selection colors
	SelectForeground: lipgloss.Color("#f0f6fc"), // Bright text
	SelectBackground: lipgloss.Color("#264f78"), // Selection blue

	// UI elements
	Border:  lipgloss.Color("#30363d"), // Border
	Link:    lipgloss.Color("#58a6ff"), // Blue
	Spinner: lipgloss.Color("#bc8cff"), // Purple

	// Tab colors
	TabActiveForeground:   lipgloss.Color("#f0f6fc"), // Bright text
	TabActiveBackground:   lipgloss.Color("#58a6ff"), // Blue
	TabInactiveForeground: lipgloss.Color("#8b949e"), // Muted text
}

// Retro theme - Matrix-inspired green phosphor on black
var retroTheme = Theme{
	Name: "retro",

	// Matrix green phosphor palette
	Primary:   lipgloss.Color("#00ff41"), // Bright matrix green
	Secondary: lipgloss.Color("#00cc33"), // Medium green
	Accent:    lipgloss.Color("#39ff14"), // Neon green

	// Status colors
	Success: lipgloss.Color("#00ff41"), // Bright green
	Warning: lipgloss.Color("#ccff00"), // Yellow-green
	Error:   lipgloss.Color("#ff003c"), // Red (digital alarm)
	Info:    lipgloss.Color("#00cc33"), // Medium green

	// Background colors
	Background:       lipgloss.Color("#0a0a0a"), // Near-black
	BackgroundAlt:    lipgloss.Color("#050505"), // Deeper black
	BackgroundSelect: lipgloss.Color("#003300"), // Dark green

	// Foreground colors
	Foreground:      lipgloss.Color("#00ff41"), // Matrix green
	ForegroundMuted: lipgloss.Color("#336633"), // Dim green
	ForegroundBold:  lipgloss.Color("#66ff66"), // Bright green

	// Selection colors
	SelectForeground: lipgloss.Color("#0a0a0a"), // Black
	SelectBackground: lipgloss.Color("#00ff41"), // Matrix green

	// UI elements
	Border:  lipgloss.Color("#004400"), // Dark green border
	Link:    lipgloss.Color("#39ff14"), // Neon green
	Spinner: lipgloss.Color("#00ff41"), // Matrix green

	// Tab colors
	TabActiveForeground:   lipgloss.Color("#0a0a0a"), // Black
	TabActiveBackground:   lipgloss.Color("#00ff41"), // Matrix green
	TabInactiveForeground: lipgloss.Color("#336633"), // Dim green
}

// Monokai theme - classic Monokai color scheme
var monokaiTheme = Theme{
	Name: "monokai",

	// Monokai palette
	Primary:   lipgloss.Color("#66d9ef"), // Cyan - headers, active status
	Secondary: lipgloss.Color("#a6e22e"), // Green - secondary info
	Accent:    lipgloss.Color("#f92672"), // Pink/Red - keybindings, section titles

	// Status colors
	Success: lipgloss.Color("#a6e22e"), // Green
	Warning: lipgloss.Color("#e6db74"), // Yellow
	Error:   lipgloss.Color("#f92672"), // Red/Pink
	Info:    lipgloss.Color("#66d9ef"), // Cyan

	// Background colors
	Background:       lipgloss.Color("#272822"), // Monokai bg
	BackgroundAlt:    lipgloss.Color("#1e1f1c"), // Darker bg
	BackgroundSelect: lipgloss.Color("#49483e"), // Selection/current line

	// Foreground colors
	Foreground:      lipgloss.Color("#f8f8f2"), // Main text
	ForegroundMuted: lipgloss.Color("#75715e"), // Comments/muted
	ForegroundBold:  lipgloss.Color("#f8f8f2"), // Emphasized text

	// Selection colors
	SelectForeground: lipgloss.Color("#f8f8f2"), // Foreground
	SelectBackground: lipgloss.Color("#49483e"), // Current line

	// UI elements
	Border:  lipgloss.Color("#75715e"), // Comment color for borders
	Link:    lipgloss.Color("#66d9ef"), // Cyan
	Spinner: lipgloss.Color("#ae81ff"), // Purple

	// Tab colors
	TabActiveForeground:   lipgloss.Color("#f8f8f2"), // Foreground
	TabActiveBackground:   lipgloss.Color("#f92672"), // Pink
	TabInactiveForeground: lipgloss.Color("#75715e"), // Muted
}
